.github/workflows/_snap-release.yaml

```diff
@@ -0,0 +1,79 @@
+on:
+  workflow_call:
+    inputs:
+      track:
+        description: |
+          The track to which charm should be released. This input supports a
+          comma-separated tracks in string format. The channels will be a list of
+          track/risk-level combinations. By default, the snap will be released to
+          channel(s) in the "latest" track.
+          Examples:
+          - "latest"
+          - "latest,test,2.9"
+        required: false
+        type: string
+        default: latest
+      risk-level:
+        description: |
+          The risk level to which charm should be released. Valid inputs are one or
+          a combination (in comma-separated string format) of: "edge", "beta",
+          "candidate", "stable". The channels will be a list of track/risk-level
+          combinations. By default, the snap will be released to channel(s) with
+          "edge" risk level.
+          Examples:
+          - "edge"
+          - "edge,candidate"
+        required: false
+        type: string
+        default: edge
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Publish Snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: snap-artifact
+      - id: get-snap-name
+        run: echo "SNAP_FILE=$(ls *.snap)" >> $GITHUB_OUTPUT
+      - name: Generate a comma separated list of channels to release the snap to
```

Comment:
Thanks for the suggestion. I was using a single `channel` input initially, but changed to `track` and `risk-level` because it reduces a bit of convolution on the source repo side. But now that you mentioned it, I think the flexibility (and simplicity of removing the combination calculation) indeed weights more than the a bit more convolution added. 

I now replaced the inputs to `channel` and added explanations in the source repo (e.g. [prometheus-juju-exporter](https://github.com/canonical/prometheus-juju-exporter/blob/f8e8fb32a6370efb674ef7edb87a0b2a87e114d0/.github/workflows/release.yaml#L25)).